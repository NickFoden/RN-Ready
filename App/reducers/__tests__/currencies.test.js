import reducer from "../currencies";
import { getInitialConversion } from "../../actions/currencies";

const initialState = {
  baseCurrency: "USD",
  quoteCurrency: "GBP",
  amount: 100,
  conversions: {},
  error: null
};

it("Sets initial state", () => {
  const expected = { ...initialState };
  const actual = reducer(undefined, {});
  expect(actual).toEqual(expected);
});

it("sets nested data on initial fetch", () => {
  const expected = {
    ...initialState,
    conversions: {
      USD: {
        date: "",
        isFetching: true,
        rates: {}
      }
    }
  };
  const actual = reducer(undefined, getInitialConversion());
  expect(actual).toEqual(expected);
});
