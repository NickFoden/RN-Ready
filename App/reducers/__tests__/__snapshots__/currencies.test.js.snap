// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets initial state with snapshot check 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "GBP",
}
`;

exports[`sets nested data on initial fetch with snapshot check 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {
    "USD": Object {
      "date": "",
      "isFetching": true,
      "rates": Object {},
    },
  },
  "error": null,
  "quoteCurrency": "GBP",
}
`;
